src/instructlab/sdg/blocks/llmblock.py

```diff
@@ -300,6 +290,13 @@ def __init__(
         parser_kwargs={},
         batch_kwargs={},
     ) -> None:
+        assert (
+            config_paths
+        ), "ConditionalLLMBlock config_paths requires at least one entry"
+        for config_path in config_paths:
+            assert (
+                len(config_path) == 2
+            ), "ConditionalLLMBlock config_paths each entry should be a list of config path and selector column names"
```

Comment:
What's the motivation to use `assert` here instead of raising?

Reply:
We typically only use `assert` outside of tests to validate/document code assumptions that are not meant to be user-facing or influenced by user config or params. However, in this case, I think you're properly catching that these config_paths are in fact user-facing (as they specify them in their pipeline yaml) and thus these should be a raised exception instead. Will fix :smile:

Reply:
This now raises a `BlockConfigParserError` instead of asserting, and adds a test to validate that gets raised properly. Thanks for the input!